cmd/flags.go

```diff
@@ -11,41 +11,43 @@ import (
 )
 
 const (
-	DefaultLayersDir     = "/layers"
-	DefaultAppDir        = "/workspace"
-	DefaultBuildpacksDir = "/cnb/buildpacks"
-	DefaultPlatformDir   = "/platform"
-	DefaultOrderPath     = "/cnb/order.toml"
-	DefaultGroupPath     = "./group.toml"
-	DefaultStackPath     = "/cnb/stack.toml"
-	DefaultAnalyzedPath  = "./analyzed.toml"
-	DefaultPlanPath      = "./plan.toml"
-	DefaultProcessType   = "web"
-	DefaultLauncherPath  = "/cnb/lifecycle/launcher"
-	DefaultLogLevel      = "info"
-
-	EnvLayersDir         = "CNB_LAYERS_DIR"
-	EnvAppDir            = "CNB_APP_DIR"
-	EnvBuildpacksDir     = "CNB_BUILDPACKS_DIR"
-	EnvPlatformDir       = "CNB_PLATFORM_DIR"
-	EnvAnalyzedPath      = "CNB_ANALYZED_PATH"
-	EnvOrderPath         = "CNB_ORDER_PATH"
-	EnvGroupPath         = "CNB_GROUP_PATH"
-	EnvStackPath         = "CNB_STACK_PATH"
-	EnvPlanPath          = "CNB_PLAN_PATH"
-	EnvUseDaemon         = "CNB_USE_DAEMON"       // defaults to false
-	EnvUseHelpers        = "CNB_USE_CRED_HELPERS" // defaults to false
-	EnvRunImage          = "CNB_RUN_IMAGE"
-	EnvCacheImage        = "CNB_CACHE_IMAGE"
-	EnvCacheDir          = "CNB_CACHE_DIR"
-	EnvLaunchCacheDir    = "CNB_LAUNCH_CACHE_DIR"
-	EnvUID               = "CNB_USER_ID"
-	EnvGID               = "CNB_GROUP_ID"
-	EnvRegistryAuth      = "CNB_REGISTRY_AUTH"
-	EnvSkipLayers        = "CNB_ANALYZE_SKIP_LAYERS" // defaults to false
-	EnvProcessType       = "CNB_PROCESS_TYPE"
-	EnvProcessTypeLegacy = "PACK_PROCESS_TYPE" // deprecated
-	EnvLogLevel          = "CNB_LOG_LEVEL"
+	DefaultLayersDir           = "/layers"
+	DefaultAppDir              = "/workspace"
+	DefaultBuildpacksDir       = "/cnb/buildpacks"
+	DefaultPlatformDir         = "/platform"
+	DefaultOrderPath           = "/cnb/order.toml"
+	DefaultGroupPath           = "./group.toml"
+	DefaultStackPath           = "/cnb/stack.toml"
+	DefaultAnalyzedPath        = "./analyzed.toml"
+	DefaultPlanPath            = "./plan.toml"
+	DefaultProcessType         = "web"
+	DefaultLauncherPath        = "/cnb/lifecycle/launcher"
+	DefaultLogLevel            = "info"
+	DefaultProjectMetadataPath = "./project-metadata.toml"
```

Comment:
Given that this metadata is supplied to the lifecycle by the platform, I would advocate that it should live in `/platform/project-metadata.toml` by default. Organizing data supplied by the platform is the intended purpose of the platform directory. This would be analogous to the default location of env vars supplied by the platform (`/platform/env`).

Given that the platform dir is currently configurable, I don't think this file path needs to be separately configurable.

When you see other files have defaults like `./group.toml`, those are files written by one lifecycle phase for consumption by another. These files end up in the `/layers` dir. Currently the platform does not modify the layers dir and I think we should stick with that convention.

Reply:
Thanks @ekcasey 
We've removed the flag for the project metadata file path, and instead expecting it at `project-metadata.toml` in the Platform directory.  Hopefully we've interpreted your comments correctly, do let us know if any other changes are required.